Add new Corkscrew Roller Coaster track pieces

mixiate opened the pull request:
This adds new pieces to the Corkscrew Roller Coaster.

Included are:
- Small flat to steep
- Vertical slopes
- Vertical turns
- Banked slope transitions
- Small banked sloped curves
- Medium banked sloped curves
- Large sloped curves
- Large banked sloped curves
- Small banked to unbanked curves
- Large corkscrews
- Quarter loops
- Medium half loops
- Large half loops
- Barrel rolls
- Zero g rolls
- Large zero g rolls

![allcorkscrewpieces](https://github.com/user-attachments/assets/e9772990-6753-4b08-95f0-a93bf5f649e5)

The large sloped curves, small banked to unbanked curves, large corkscrews, medium and large half loops are made available to the corkscrew roller coaster. The banked sloped curves are available for trains with those sprites. All other pieces are available through cheats only.

I've done my best to make all the pieces function correctly. I've tested with the RCT 1 and 2 trains, [spaceks modern corkscrew trains](https://github.com/spacek531/custom-rides/raw/master/Modern%20Corkscrew%20CTR.zip), and a test train I made that has roughly the same dimensions as the RCT 2 train.
Some compromises were made because it was quite difficult to get the corkscrew train to go through some of these pieces without glitching.

I've also added the code to change the version and make pre existing pieces invisible.

SpartanFrederic104 commented:
![Fred's Workbench 2024-10-08 22-41-59](https://github.com/user-attachments/assets/34d80259-7e4c-498b-9270-2e3a7319fe7b)
Tested these new track pieces and so far they look insanely nice. Large half corkscrews and larger loops are always welcomed, especially in a coaster type that tends to be very intense partly due to the RCT2 trains giving it a +10% intensity boost (the RCT1 trains don't do this) and the relatively high G-forces from entering regular corkscrews.

![Fred's Workbench 2024-10-08 22-43-16](https://github.com/user-attachments/assets/91e8e4ad-85ca-4e04-8d3b-b8ac811a368a)
If I had any issues with this, it would be a relatively small issue with the lighting/shading at the start/end of the medium zero-G roll piece in this angle pictured above (it instantly goes from bright to a slightly darkened hue as soon as the zero-G roll starts/ends). Other than that, the new track pieces look great and I can't wait to see these new track pieces in action in the official builds soon enough.

xrgtmx commented:
Needs changelog, network, and park tags.

mixiate commented:
Thanks for testing out these pieces again!

I haven't commited it yet but I tried to lighten the zero g roll a little bit without completely removing the shadow.

![newzerogroll](https://github.com/user-attachments/assets/9e8b4132-6d0a-49ba-8809-1b18c70d25d6)

Let me know what you think.

SpartanFrederic104 commented:
That looks better now.

Gymnasiast commented:
I’ll look at this one after the Wooden RC loops PR has been merged.

Gymnasiast commented:
Could you rebase this one?

mixiate commented:
I've rebased this. Apologies for the several pushes, i'm not getting these warnings/errors when compiling myself for some reason. I will see if I can do anything about that.

I hope it's okay but I also added a commit to make the curves available for the lay down coaster, and added it to the list for invisible pieces. Sorry I didn't catch this before.

Gymnasiast commented on src/openrct2/paint/track/coaster/CorkscrewRollerCoaster.cpp:
```diff
@@ -11046,6 +11046,8353 @@ static void LayDownRCTrackHalfLoopUninvertedDown(
+    }
+}
+
+static void CorkscrewRCTrack90DegUp(
```
There are already sprites for this on the Compact Inverted Coaster. Is there any specific reason you didn’t reuse these?

mixiate replied:
Ah I didn't even realise. I could use it if you'd like. I'd have to rerender a few pieces to join it correctly.

karst commented:
Idem on the conversion code I mentioned in another one of your PRs.

mixiate commented:
They both already have it don't they? Does it not work?

mixiate commented:
This should be okay now if you wanted to take another look at it.
